### ISSUE TITLE
Beer Factories | Trading amounts are not correctly set after construction of new Trading Post

### ISSUE THREAD

**Issue by 2ndK16** [wol-soft/beerplop-issues#56]
I built a new trading post and the traded amount were not set to the maximum value, as they should have, but only to one half of the maximum value.

**Comment by wol-soft** [comment]
@2ndK16 was it possible to max the traded amount with the MAX button to the expected value afterwards? At which levels are your trading post upgrade paths?

**Comment by 2ndK16** [comment]
@wol-soft yes, setting the value to MAX manually worked fine. I have 20 TPs on level 4/2.

**Comment by wol-soft** [comment]
@2ndK16 the option to automatically update the trading amount to the maximum value is enabled for each route?

I just tried to reproduce the issue without success. The amount was updated correctly. I also tried to reproduce it with an active Beer Factory boost but the scenario also worked for me. The values were updated and a notification about the automatic update was shown.

**Comment by 2ndK16** [comment]
I've ticked all three checkboxes and it also happens whenever I reload the page.

**Comment by wol-soft** [comment]
What happens when you reload the page? A wrong amount is calculated and configured for the trading routes?

**Comment by 2ndK16** [comment]
The following happens:
I trade 640 Diamond in exchange for 457 Tools on each of the three routes.
I refresh the page.
All three routes are set to 320 Diamond for 228 Tools.
I set the trading amount to MAX manually.
I trade 640 Diamond in exchange for 457 Tools on each of the three routes.

**Comment by wol-soft** [comment]
Can you provide the name of your save state so I can have a look into it? As described above I'm currently not able to reproduce it but it seems to be reproducable with your save state.

**Comment by 2ndK16** [comment]
My current save state is named TimeThief7 by user ysc

**Comment by wol-soft** [comment]
Thanks, I'll have a look into it

**Comment by wol-soft** [comment]
The traded amount is recalculated during reload to avoid too high numbers caused by saving during a Beer Factory Buff (#49). Currently the Pyramids of Giza equipment of the Was Scepter wasn't calculated correctly during this recalculation as it hasn't been fully loaded. I've moved the recalculation so it's not executed until the complete save state is loaded.

I've released the patch and a reload now gives me the correct amount of 640 diamonds - 457 tools.

If I construct a new trading post the values are updated (which was the initial issue, so I'm not sure if the issue is resolved completely)
